likecoin/admin/restful.php

```diff
@@ -45,6 +45,202 @@ function likecoin_rest_refresh_publish_status( $request ) {
+	$likecoin_api_estimate_url = 'https://api.like.co/api/arweave/estimate'; // TODO: change endpoint based on test/production.
+	$response                  = wp_remote_post(
+		$likecoin_api_estimate_url,
+		array(
+			'headers' => array(
+				'Content-Type' => 'multipart/form-data; boundary=' . $boundary,
+			),
+			'body'    => $body,
+		)
+	);
+	if ( is_wp_error( $response ) ) {
+		$err_message = $response->get_error_message();
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			error_log( $response->get_error_message() );
+		}
+		return array( 'error' => $response->get_error_message() );
+	}
+	$decoded_response = json_decode( $response['body'], true );
+	if ( ! $decoded_response ) {
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			error_log( $response['body'] );
+		}
+		return array( 'error' => $response['body'] );
+	}
+	$data = json_decode( $response['body'], true );
+	return new WP_REST_Response( $data, 200 );
+}
+
+/**
+ * Add likecoin arweave upload endpoint.
+ *
+ * @param WP_REST_Request $request Full data about the request.
+ * @return WP_Error|WP_REST_Response
+ */
+function likecoin_rest_post_arweave_upload( $request ) {
+	$post_id = $request['id'];
+	$post    = get_post( $post_id );
+	$tx_hash = $request['txHash'];
+	if ( ! isset( $post ) ) {
+		return new WP_Error( 'post_not_found', __( 'Post was not found', LC_PLUGIN_SLUG ), array( 'status' => 404 ) );
+	}
+	$content        = likecoin_filter_matters_post_content( $post );
+	$urls           = likecoin_get_post_image_url( $post );
+	$content        = '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
+        "http://www.w3.org/TR/html4/strict.dtd"><html><body>' . $content . '</body></html>';
+	$file_mime_type = 'text/html';
+	$filename       = 'index.html';
+	// phpcs:disable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$boundary = base64_encode( wp_generate_password( 24 ) );
+	// phpcs:enable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$body  = '';
+	$body .= '--' . $boundary . "\r\n";
+	$body .= 'Content-Disposition: form-data; name="index.html"; filename="' . $filename . "\"\r\n";
+	$body .= 'Content-Type: ' . $file_mime_type . "\r\n";
+	$body .= "Content-Transfer-Encoding: binary\r\n";
+	$body .= "\r\n";
+	$body .= $content . "\r\n";
+	$body .= "\r\n";
+	foreach ( $urls as $url ) {
+		$file_info  = new finfo( FILEINFO_MIME_TYPE );
+		$image_path = substr( $url, 22 );
+		$img_body   = file_get_contents( $image_path );
+		$mime_type  = $file_info->buffer( $img_body );
+		$filename   = basename( $url );
+		$body      .= '--' . $boundary . "\r\n";
+		$body      .= 'Content-Disposition: form-data; name="' . $filename . '"; filename="' . $filename . "\"\r\n";
+		$body      .= 'Content-Type: ' . $mime_type . "\r\n";
+		$body      .= "Content-Transfer-Encoding: binary\r\n";
+		$body      .= "\r\n";
+		$body      .= $img_body . "\r\n";
+		$body      .= "\r\n";
+	}
+	$body                   .= '--' . $boundary . '--';
```

Comment:
Is this part identical with the above? If yes please refactor to a function, since this part indeed is very long and hard to read